cider/tests/nextgen/jitlib/JITLibTest.cpp

```diff
@@ -501,6 +501,84 @@ TEST_F(JITLibTests, NestedForControlFlowTest) {
   EXPECT_EQ(func_ptr(), 100000);
 }
 
+TEST_F(JITLibTests, ReadOnlyJITPointerTest) {
+  LLVMJITModule module("TestModule");
+  JITFunctionPointer function = module.createJITFunction(JITFunctionDescriptor{
+      .function_name = "test_func",
+      .ret_type = JITFunctionParam{.name = "ret", .type = JITTypeTag::INT32},
+      .params_type = {JITFunctionParam{.name = "array",
+                                       .type = JITTypeTag::POINTER,
+                                       .sub_type = JITTypeTag::INT32},
+                      JITFunctionParam{.name = "len", .type = JITTypeTag::INT32}}});
+  {
+    JITValuePointer ret = function->createVariable("ret", JITTypeTag::INT32);
+    ret = function->createConstant(JITTypeTag::INT32, 0);
+
+    auto loop_builder = function->createLoopBuilder();
+    JITValuePointer index = function->createVariable("index", JITTypeTag::INT32);
+
+    auto array_ptr = function->getArgument(0);
+    auto array_len = function->getArgument(1);
+    *index = array_len - 1;
+
+    loop_builder->condition([&]() { return index + 0; })
```

Comment:
does `return index;` work?

Reply:
I found the unique_ptr made lots of inconvenience, maybe we could manage JIValuePointer by a simple reference counter, your thoughts?

Reply:
> I found the unique_ptr made lots of inconvenience

agree with you absolutely.
`JITValuePointer` has no other benefits except ensuring one `JITValue` only can be used once, which is ssa.